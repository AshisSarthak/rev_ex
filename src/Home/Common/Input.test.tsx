import React from "react";
import { shallow } from "enzyme";
import InputEl from "./Input";

describe("Input Element", () => {
  const mockFn = jest.fn();

  it("Loads the default Dropdown", () => {
    const wrapper = shallow(<InputEl value={10} onChange={mockFn} disabled />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.prop("placeholder")).toBe("Enter Amount");
    wrapper.simulate("change", {
      target: {
        value: 15,
      },
    });
    expect(mockFn).toBeCalledWith(15);
  });
});
