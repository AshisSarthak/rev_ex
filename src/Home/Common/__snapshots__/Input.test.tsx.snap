// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Element Loads the default Dropdown 1`] = `
<input
  className="exchange-from-input"
  disabled={true}
  onChange={[Function]}
  placeholder="Enter Amount"
  type="number"
  value={10}
/>
`;
